ansible_collections/arista/avd/roles/eos_designs/templates/designs/l3ls-evpn/facts/switch/switch.j2

```diff
@@ -305,7 +305,7 @@ switch:
 {% set leaf = namespace() %}
 {% set leaf.vlans = []  %}
 {% set leaf.tenants = {} %}
-{% for tenant in tenants | arista.avd.natural_sort if tenant in switch_filter_tenants or "all" in switch_filter_tenants %}
+{% for tenant in tenants | arista.avd.natural_sort if switch_filter_tenants is arista.avd.defined and ( tenant in switch_filter_tenants or "all" in switch_filter_tenants) %}
```

Comment:
Since this variable is dependant on two other variables, I suggest that we change this to test for those two above the whole vlan/vrf section instead (ex. line 304)
```
{% if switch.network_services_l2 is arista.avd.defined(true) or
      switch.network_services_l3 is arista.avd.defined(true) %}
```

Reply:
Code updated as per your review.
Also added a `arista.avd.default` in topology facts to avoid error message for this change